cristin-commons/src/main/java/no/unit/nva/cristin/common/Utils.java

```diff
@@ -40,4 +43,12 @@ public static boolean isPositiveInteger(String str) {
     public static boolean isOrcid(String identifier) {
         return Constants.ORCID_PATTERN.matcher(identifier).matches();
     }
+
+    /**
+     * A function that can be used to filter out duplicate values based on a given key.
+     */
+    public static <T> Predicate<T> distinctByKey(Function<? super T, ?> keyExtractor) {
+        Map<Object, Boolean> seen = new ConcurrentHashMap<>();
```

Comment:
A question about posting pre-existing NINs to Cristin: does Cristin pre-filter existing records with the same NIN, or can multiple records have the same NIN (not sure how the model is…)?

Reply:
NIN must be unique to one person if I remember correctly